std/mmfile.d

```diff
@@ -369,15 +372,24 @@ class MmFile
         }
         else version (Posix)
         {
-            // enforce shouldn't be here and close() is safe anyway
-            .close(fd);
+            version (linux)
+            {
+                if (file !is File.init)
+                {
+                    // The File destructor will close the file,
+                    // if it is the only remaining reference.
+                    return;
+                }
+            }
+            errnoEnforce(fd == -1 || fd <= 2
```

Comment:
That wouldn't be an improvement. `enforce` will have to do. Let's not forget that class destructors are not always invoked by the GC (there's e.g. `scoped`), so removing `enforce` would penalize uses that do not use the GC.